integration-tests/smoke/ccip_messaging_test.go

```diff
@@ -165,18 +171,159 @@ func Test_CCIPMessaging(t *testing.T) {
 			},
 			state.Chains[destChain].Receiver.Address(),
 			[]byte("hello CCIPReceiver with low exec gas"),
-			ccipdeployment.MakeEVMExtraArgsV2(1, false), // 1 gas is too low.
-			ccipdeployment.EXECUTION_STATE_FAILURE,      // state would be failed onchain due to low gas
+			ccdeploy.MakeEVMExtraArgsV2(1, false), // 1 gas is too low.
+			ccdeploy.EXECUTION_STATE_FAILURE,      // state would be failed onchain due to low gas
 		)
+
+		manuallyExecute(ctx, t, state, destChain, out, sourceChain, e, sender)
+
+		t.Logf("successfully manually executed message %x",
+			out.msgSentEvent.Message.Header.MessageId)
 	})
 }
 
-func sleepAndReplay(t *testing.T, e ccipdeployment.DeployedEnv, sourceChain, destChain uint64) {
+func manuallyExecute(
```

Comment:
is it possible to extract the manual execute logic as a generic function which can be used from deployment package ?

Reply:
There isn't much to extract beyond the merklemulti utils which are already extracted. We have similar login in the plugin for [exec](https://github.com/smartcontractkit/chainlink-ccip/blob/b53930a1bdea5962151ad242622cdf5cf3813487/execute/report/roots.go#L16) and [commit ](https://github.com/smartcontractkit/chainlink-ccip/blob/b53930a1bdea5962151ad242622cdf5cf3813487/commit/merkleroot/observation.go#L417), unfortunately even there it was difficult to reuse the helper.

Reply:
Yeah, I want to make this more generic, but there are a few challenges:

1. Similar to the exec plugin, you need to get the commit report corresponding to a particular message, and then fetch all the messages that made up that root
2. Once you fetch all the messages, you need to hash them all and construct the tree leaves
3. Once you get all the tree leaves, you can use `merklemulti.NewTree` to construct the tree and the proof

This function is overfit to the one message per root case which is why its surprisingly straightforward. The more general case would be more complex. Step (1) in particular is quite bad because the `CommitReportAccepted` log has nothing that we can use to narrow down the log query, so you have to fetch all the logs and do an in-memory filter (with the absence of something more structured like Atlas databases).

Reply:
I think I'll do this in a follow-up where I'd want to test manual exec for a multi-message root.

Reply:
One related thing that I think we should be doing is that our log queries shouldn't just be passing in `nil` for the `FilterOpts` (I am doing this right now but it's wrong) and we also shouldn't be replaying from block 1. This makes these tests not really runnable on testnets. I think we'd have to do a sweep on all of these and make sure we catch all the stragglers.